legacy/reqcounter/reqcounter_test.go

```diff
@@ -124,3 +126,57 @@ func TestRequestCounterIncrement(t *testing.T) {
 	// Ensure the request counter was incremented.
 	require.EqualValues(t, &expected, &requestCounter, "The request counter failed to increment its request field.")
 }
+
+func TestLog(t *testing.T) {
+	// Create multiple request counters with unique number of starting requests. These unique requests will
+	// be the way to differentiate between request counter log statements.
+	requestCounters := []rc.RequestCounter{
+		NewRequestCounter(21),
+		NewRequestCounter(9),
+		NewRequestCounter(2839),
+	}
+	netMonitor := &rc.NetworkMonitor{
+		RequestCounters: rc.RequestCounters{
+			&requestCounters[0],
+			&requestCounters[1],
+			&requestCounters[2],
+		},
+	}
+	// Mock time with a fake global time that remains constant. This avoids testing the order in which
+	// request counters are logged. For simplicity, additional tests may check whether a single request
+	// counter calls Debug() the expected number of times as fake time advances forward. This circumvents
+	// the task of concurrently communicating time-steps with multiple goroutines, demanding a
+	// complicated sleep and time-step synchronization.
+	rc.Clock = tw.FakeTime{Time: defaultTime}
+	// Record the unique logging statements from each request counter.
+	// Block goroutines after reset(), where all logging messages will be identical.
+	logged := map[string]bool{}
+	stableTime := defaultTime.Format(rc.TimestampFormat)
+	// Craft the expected log which will block.
+	resetLog := fmt.Sprintf("From %s to %s [0 min] there have been 0 requests to GCR.", stableTime, stableTime)
+	// Use a wait group to sure each request counter has finished recording their debug statements.
+	wg := sync.WaitGroup{}
+	wg.Add(len(requestCounters))
+	// Mock the logrus.Debug function.
+	rc.Debug = func(args ...interface{}) {
+		msg := fmt.Sprint(args[0])
+		if resetLog == msg {
+			// Block the current goroutine.
+			select {}
+		}
+		logged[msg] = true
+		wg.Done()
+	}
+	// Define the expected logs.
+	expected := map[string]bool{
+		fmt.Sprintf("From %s to %s [%d min] there have been %d requests to GCR.", stableTime, stableTime, requestCounters[0].Interval/time.Minute, requestCounters[0].Requests): true,
```

Comment:
here and elsewhere all "expected" strings should be literals.

don't use requestCounters[0].Requests use the literal value that you expect to see.